journey-map/public/render.mjs

```diff
@@ -70,73 +26,69 @@ const expandParents = (journeyStates, otherStates) => {
             Object.entries(subjourney.nestedJourneyStates).forEach(([nestedState, nestedDefinition]) => {
                 // Copy to avoid mutating different versions of the expanded definition
                 const expandedDefinition = JSON.parse(JSON.stringify(nestedDefinition));
 
-                Object.entries(expandedDefinition.events || {}).forEach(([evt, def]) => {
-                    // Map target states to expanded states
-                    if (def.targetState && !def.targetJourney) {
-                        def.targetState = `${def.targetState}_${state}`;
-                    }
+                Object.entries(expandedDefinition.events || {}).forEach(([evt, eventDef]) => {
+                    mapTargetStateToExpandedState(eventDef, subJourneyState);
 
                     // Map exit events to targets in the parent definition
-                    const exitEvent = def.exitEventToEmit;
+                    const exitEvent = eventDef.exitEventToEmit;
                     if (exitEvent) {
-                        delete def.exitEventToEmit;
+                        delete eventDef.exitEventToEmit;
                         if (definition.exitEvents[exitEvent]) {
-                            Object.assign(def, definition.exitEvents[exitEvent]);
+                            Object.assign(eventDef, definition.exitEvents[exitEvent]);
                         } else {
-                            console.warn(`Unhandled exit event from ${state}:`, exitEvent)
+                            console.warn(`Unhandled exit event from ${subJourneyState}:`, exitEvent)
                             delete expandedDefinition.events[evt];
                         }
                     }
                 });
 
-                journeyMap[`${nestedState}_${state}`] = expandedDefinition;
+                journeyMap[`${nestedState}_${subJourneyState}`] = expandedDefinition;
             });
 
+            // Make a copy of the event definition to avoid mutating the original
+            const entryEvents = JSON.parse(JSON.stringify(subjourney.entryEvents));
             // Update entry events on other states to expanded states
-            Object.entries(subjourney.entryEvents).forEach(([entryEvent, def]) => {
+            Object.entries(entryEvents).forEach(([entryEvent, entryEventDef]) => {
+                mapTargetStateToExpandedState(entryEventDef, subJourneyState);
+
                 Object.values(journeyMap).forEach((journeyDef) => {
                     if (journeyDef.events?.[entryEvent]) {
-                        const target = resolveEventTarget(journeyDef.events[entryEvent], formData);
-                        if (target.targetState === state && !target.targetEntryEvent) {
-                            target.targetState = `${def.targetState}_${state}`;
+                        const target = resolveEventTargets(journeyDef.events[entryEvent], undefined, formData).find(t => !t.journeyContext)
```

Comment:
`resolveTargets` atm only returns multiple targets if there is a journeyContext in the definition (since we only replace targets when checking feature flags or when CRIs are disabled) so we take the first target definition without a journeyContext

Reply:
I'm not totally sure I follow this - don't we (potentially) need to replace targets with a journeyContext as well?

Not too worried if you want to make this a bit of a simplification though - I pretty much never use this option anyway!

Reply:
From what I understand from before, this bit is just updating the targetStates of the entry events to a mapped subjourney target value (e.g. `targetState: TARGET` => `targetState: TARGET_SUBJOURNEY` to display which states are part of a subjourney, in which case we don't need to worry about replacing with a journeyContext bc mapping of all nestedJourneyStates is done above (from line 50) and we handle the transitions between these (inc when there is a journeyContext) in `renderTransitions`